contracts/ClaimsManager.sol

```diff
@@ -287,6 +292,51 @@ contract ClaimsManager is
         );
     }
 
+    function downgradePolicy(
+        address claimant,
+        address beneficiary,
+        uint256 coverageAmountInUsd,
+        uint256 claimsAllowedFrom,
+        uint256 claimsAllowedUntil,
+        string calldata policy,
+        string calldata metadata
+    ) external override returns (bytes32 policyHash) {
+        require(claimant == msg.sender, "Sender not claimant");
+        policyHash = keccak256(
+            abi.encodePacked(
+                claimant,
+                beneficiary,
+                claimsAllowedFrom,
+                policy,
+                metadata
+            )
+        );
+        PolicyState storage policyState = policyHashToState[policyHash];
+        require(policyState.claimsAllowedUntil != 0, "Policy does not exist");
+        require(
+            policyState.coverageAmountInUsd >= coverageAmountInUsd,
+            "Increases coverage amount"
+        );
+        require(
+            policyState.claimsAllowedUntil >= claimsAllowedUntil,
+            "Allows claims for longer"
+        );
+        policyHashToState[policyHash] = PolicyState({
+            claimsAllowedUntil: uint32(claimsAllowedUntil),
+            coverageAmountInUsd: uint224(coverageAmountInUsd)
+        });
+        emit DowngradedPolicy(
```

Comment:
Hmm is the reasoning for having two events (`UpgradedPolicy` and `DowngradedPolicy`) vs a single event (like `UpdatedPolicy`) so that the tracker can distinguish between the two?

Reply:
I'm just being verbose. You can deduce if it's an upgrade or a downgrade based on the previous state but feels like that's asking for too much. Also the upgrade has a sender field (which you can get from the tx hash with a second provider call), while the downgrade doesn't because the sender is guaranteed to be the claimant and having a sender field would have been confusing.

Reply:
I should be able to get by. FYI I currently use the `UpgradedPolicy` log just to get the latest `claimsAllowedUntil` upfront (and I use the policyState to get `coverageAmountInUsd` just for the current page of policies the user is viewing)

Reply:
Fixed in https://github.com/api3dao/claims-manager/pull/55/commits/3391f023a56a43c649e8ce060f103aaed60acf74